### PR TITLE
ResourceQuota BestEffort scope aligned with Pod level QoS

### PR DESCRIPTION
**Pull request by derekwaynecarr**
This aligns quota with the changes in kubelet and CLI.

So if quota allows 10 `BestEffort` pods, it will now track properly with what the user sees with changes in 1.3.

```
apiVersion: v1
kind: ResourceQuota
metadata:
  name: best-effort
spec:
  hard:
    pods: "10"
  scopes:
  - BestEffort
```

/cc @vishh @kubernetes/rh-cluster-infra

### CODE REVIEW

File: pkg/quota/evaluator/core/pods.go [hunk 1]
```diff
@@ -172,16 +172,7 @@ func PodMatchesScopeFunc(scope api.ResourceQuotaScope, object runtime.Object) bo
...
-	for _, container := range pod.Spec.Containers {
-		qosPerResource := util.GetQoS(&container)
-		for _, qos := range qosPerResource {
-			if util.BestEffort == qos {
-				return true
-			}
-		}
-	}
-	return false
+	return util.GetPodQos(pod) == util.BestEffort
```
Comment (dims) [inline pkg/quota/evaluator/core/pods.go]: Newbie question : GetPodQos seems to return either Burstable or Guaranteed or BestEffort. So here we look for an exact match for util.BestEffort? (in other words, does ResourceQuotaScopeBestEffort, should it be matched with both Guaranteed and BestEffort)?

**Comment by vishh** [comment]
LGTM

File: pkg/quota/evaluator/core/pods.go (hunk 1 above)
Comment (derekwaynecarr) [inline pkg/quota/evaluator/core/pods.go]: https://github.com/kubernetes/kubernetes.github.io/pull/622

This is an advanced feature of resourcequota.  See step 5 in PR above that discusses scopes filtering the set of resources a quota tracks in an example.

Full doc on scopes and quota:
https://github.com/kubernetes/kubernetes.github.io/pull/620

Comment (dims) [inline pkg/quota/evaluator/core/pods.go]: Thanks @derekwaynecarr

**Comment by yujuhong** [comment]
@k8s-bot node e2e test this issue: #26431

**Pull request merged**